tests/func-tests/passt_test.go

```diff
@@ -0,0 +1,169 @@
+package tests_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	netattdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/handlers/passt"
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+)
+
+const (
+	passtCNIObjectName         = "passt-binding-cni"
+	networkBindingNADName      = "primary-udn-kubevirt-binding"
+	networkBindingNADNamespace = "default"
+)
+
+var _ = Describe("Test Passt Network Binding", Label("Passt"), Serial, Ordered, func() {
+	tests.FlagParse()
+	var (
+		k8scli client.Client
+	)
+
+	BeforeEach(func(ctx context.Context) {
+		k8scli = tests.GetControllerRuntimeClient()
+
+		err := netattdefv1.AddToScheme(k8scli.Scheme())
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func(ctx context.Context) {
+		disablePasstDeployment(ctx, k8scli)
+	})
+
+	When("deployPasstNetworkBinding annotation is set to true", func() {
+		It("should create the passt DaemonSet and NetworkAttachmentDefinition", func(ctx context.Context) {
+			By("enabling passt deployment annotation")
+			enablePasstDeployment(ctx, k8scli)
+
+			By("checking that the passt DaemonSet is created and ready")
+			Eventually(func(g Gomega, ctx context.Context) {
+				ds := &appsv1.DaemonSet{}
+				err := k8scli.Get(ctx, types.NamespacedName{
+					Name:      passtCNIObjectName,
+					Namespace: tests.InstallNamespace,
+				}, ds)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(ds.Status.NumberReady).To(BeNumerically(">", 0))
+				g.Expect(ds.Status.DesiredNumberScheduled).To(Equal(ds.Status.NumberReady))
+			}).WithTimeout(5 * time.Minute).
+				WithPolling(10 * time.Second).
+				WithContext(ctx).
+				Should(Succeed())
+
+			By("checking that the passt NetworkAttachmentDefinition is created")
+			Eventually(func(g Gomega, ctx context.Context) {
+				nad := &netattdefv1.NetworkAttachmentDefinition{}
+				err := k8scli.Get(ctx, types.NamespacedName{
+					Name:      networkBindingNADName,
+					Namespace: networkBindingNADNamespace,
+				}, nad)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(nad.Spec.Config).ToNot(BeEmpty())
+			}).WithTimeout(2 * time.Minute).
+				WithPolling(5 * time.Second).
+				WithContext(ctx).
+				Should(Succeed())
+		})
+
+		It("should remove passt resources when annotation is disabled", func(ctx context.Context) {
```

Comment:
Maybe we can add it as steps to the prev test, to avoid duplications?

Reply:
aint a test need to do one thing ? 
anyhow whatever you prefer
it will just make the test do more / be longer